<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>qSlicerUltrasoundSegmentationModuleWidget</class>
 <widget class="qSlicerWidget" name="qSlicerUltrasoundSegmentationModuleWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>656</width>
    <height>857</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="ctkCollapsibleButton" name="CTKCollapsibleButton">
     <property name="text">
      <string>Tranning</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <item>
       <widget class="QGroupBox" name="groupBox">
        <property name="title">
         <string>Training file</string>
        </property>
        <layout class="QGridLayout" name="gridLayout">
         <item row="1" column="0">
          <widget class="QPushButton" name="loadFileButton">
           <property name="text">
            <string>Load Training File</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLineEdit" name="fileLineEdit"/>
         </item>
         <item row="0" column="0">
          <widget class="QCheckBox" name="useFileCheckBox">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="text">
            <string>Use tranning file</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="trainingGroupBox">
        <property name="title">
         <string>Tranning</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_2">
         <item row="2" column="0">
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>Tissue Label:</string>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>Shadow Label:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="qMRMLNodeComboBox" name="sourceTrainMRMLNodeComboBox">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="nodeTypes">
            <stringlist>
             <string>vtkMRMLVolumeNode</string>
            </stringlist>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>Bone Label:</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="qMRMLNodeComboBox" name="tissueMRMLNodeComboBox">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="nodeTypes">
            <stringlist>
             <string>vtkMRMLVolumeNode</string>
            </stringlist>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="qMRMLNodeComboBox" name="shadowMRMLNodeComboBox">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="nodeTypes">
            <stringlist>
             <string>vtkMRMLVolumeNode</string>
            </stringlist>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="qMRMLNodeComboBox" name="boneMRMLNodeComboBox">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="nodeTypes">
            <stringlist>
             <string>vtkMRMLVolumeNode</string>
             <string></string>
            </stringlist>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QLabel" name="label">
           <property name="text">
            <string>Source Volume:</string>
           </property>
          </widget>
         </item>
         <item row="6" column="0" colspan="2">
          <widget class="QFrame" name="frame">
           <property name="frameShape">
            <enum>QFrame::StyledPanel</enum>
           </property>
           <property name="frameShadow">
            <enum>QFrame::Raised</enum>
           </property>
           <property name="lineWidth">
            <number>0</number>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayout">
            <item>
             <widget class="QPushButton" name="trainButton">
              <property name="text">
               <string>Train</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="saveFileButton">
              <property name="text">
               <string>Save Tranning File</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="ctkCollapsibleButton" name="CollapsibleButton">
     <property name="text">
      <string>Segmentation</string>
     </property>
     <property name="contentsFrameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QGroupBox" name="sourceGroupBox">
        <property name="title">
         <string>Sources</string>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_5">
         <item>
          <widget class="QLabel" name="label_11">
           <property name="text">
            <string>Source Volume:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="qMRMLNodeComboBox" name="sourceSegmentMRMLNodeComboBox">
           <property name="nodeTypes">
            <stringlist>
             <string>vtkMRMLVolumeNode</string>
            </stringlist>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_3">
        <property name="title">
         <string>Structure</string>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QRadioButton" name="sheetRadioButton">
           <property name="text">
            <string>Sheet</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="tubeRadioButton">
           <property name="text">
            <string>Tube</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="blobRadioButton">
           <property name="text">
            <string>Blob</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_4">
        <property name="title">
         <string>Scale:</string>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="QLabel" name="label_5">
           <property name="text">
            <string>Min. Scale</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDoubleSpinBox" name="minSpinBox">
           <property name="minimum">
            <double>0.100000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.100000000000000</double>
           </property>
           <property name="value">
            <double>1.500000000000000</double>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QLabel" name="label_6">
           <property name="text">
            <string>Max. Scale</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDoubleSpinBox" name="maxSpinBox">
           <property name="minimum">
            <double>0.100000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.100000000000000</double>
           </property>
           <property name="value">
            <double>4.000000000000000</double>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_2">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QLabel" name="label_7">
           <property name="text">
            <string>Scales</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="scalesSpinBox">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="value">
            <number>1</number>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_5">
        <property name="title">
         <string>Enhancement</string>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <item>
          <widget class="QLabel" name="label_8">
           <property name="text">
            <string>Alpha:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDoubleSpinBox" name="alphaSpinBox">
           <property name="singleStep">
            <double>0.100000000000000</double>
           </property>
           <property name="value">
            <double>1.000000000000000</double>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_3">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QLabel" name="label_9">
           <property name="text">
            <string>Beta:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDoubleSpinBox" name="betaSpinBox">
           <property name="singleStep">
            <double>0.100000000000000</double>
           </property>
           <property name="value">
            <double>1.000000000000000</double>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_4">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QLabel" name="label_10">
           <property name="text">
            <string>Gamma:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDoubleSpinBox" name="gammaSpinBox">
           <property name="singleStep">
            <double>0.100000000000000</double>
           </property>
           <property name="value">
            <double>0.010000000000000</double>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="isBrightCheckBox">
        <property name="text">
         <string>Bright Object</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="Line" name="line">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="pushButton_3">
        <property name="text">
         <string>Segment</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>0</width>
       <height>0</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>qMRMLNodeComboBox</class>
   <extends>QWidget</extends>
   <header>qMRMLNodeComboBox.h</header>
   <slots>
    <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
    <slot>setMRMLScene(vtkMRMLScene*)</slot>
   </slots>
  </customwidget>
  <customwidget>
   <class>qSlicerWidget</class>
   <extends>QWidget</extends>
   <header>qSlicerWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ctkCollapsibleButton</class>
   <extends>QWidget</extends>
   <header>ctkCollapsibleButton.h</header>
   <container>1</container>
   <slots>
    <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>useFileCheckBox</sender>
   <signal>stateChanged(int)</signal>
   <receiver>qSlicerUltrasoundSegmentationModuleWidget</receiver>
   <slot>enableTraining()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>168</x>
     <y>85</y>
    </hint>
    <hint type="destinationlabel">
     <x>646</x>
     <y>81</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>loadFileButton</sender>
   <signal>clicked()</signal>
   <receiver>qSlicerUltrasoundSegmentationModuleWidget</receiver>
   <slot>loadTrainingFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>132</x>
     <y>117</y>
    </hint>
    <hint type="destinationlabel">
     <x>647</x>
     <y>114</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>trainButton</sender>
   <signal>clicked()</signal>
   <receiver>qSlicerUltrasoundSegmentationModuleWidget</receiver>
   <slot>train()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>278</x>
     <y>353</y>
    </hint>
    <hint type="destinationlabel">
     <x>651</x>
     <y>404</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>saveFileButton</sender>
   <signal>clicked()</signal>
   <receiver>qSlicerUltrasoundSegmentationModuleWidget</receiver>
   <slot>saveTraining()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>568</x>
     <y>353</y>
    </hint>
    <hint type="destinationlabel">
     <x>647</x>
     <y>353</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_3</sender>
   <signal>clicked()</signal>
   <receiver>qSlicerUltrasoundSegmentationModuleWidget</receiver>
   <slot>segment()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>612</x>
     <y>787</y>
    </hint>
    <hint type="destinationlabel">
     <x>650</x>
     <y>788</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>qSlicerUltrasoundSegmentationModuleWidget</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>sourceSegmentMRMLNodeComboBox</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>327</x>
     <y>428</y>
    </hint>
    <hint type="destinationlabel">
     <x>381</x>
     <y>484</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>qSlicerUltrasoundSegmentationModuleWidget</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>sourceTrainMRMLNodeComboBox</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>327</x>
     <y>428</y>
    </hint>
    <hint type="destinationlabel">
     <x>382</x>
     <y>186</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>qSlicerUltrasoundSegmentationModuleWidget</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>boneMRMLNodeComboBox</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>327</x>
     <y>428</y>
    </hint>
    <hint type="destinationlabel">
     <x>382</x>
     <y>222</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>qSlicerUltrasoundSegmentationModuleWidget</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>tissueMRMLNodeComboBox</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>327</x>
     <y>428</y>
    </hint>
    <hint type="destinationlabel">
     <x>382</x>
     <y>258</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>qSlicerUltrasoundSegmentationModuleWidget</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>shadowMRMLNodeComboBox</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>327</x>
     <y>428</y>
    </hint>
    <hint type="destinationlabel">
     <x>382</x>
     <y>294</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>enableTraining()</slot>
  <slot>loadTrainingFile()</slot>
  <slot>train()</slot>
  <slot>saveTraining()</slot>
  <slot>segment()</slot>
 </slots>
</ui>
